### PR TITLE
Add release notes for Trilinos 14.0 and break in backward compatibility with -isystem include dirs

### PR DESCRIPTION
**Pull request by bartlettroscoe**
Related TriBITS PRs are:

* TriBITSPub/TriBITS#456
* TriBITSPub/TriBITS#459

For more details, see TriBITSPub/TriBITS#443

### CODE REVIEW

**Review by marcinwrobel1986 — APPROVED** [review]
No typos, good to go!

**Comment by bartlettroscoe** [comment]
I will need to wait to merge this to my branch 'tribits-299-modern-cmake-targets-1-again' until after I updated the TriBITS snapshot in that branch.

File: RELEASE_NOTES [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+#                                                                             #
+###############################################################################
+
+CMake
+
+  - Switch to modern CMake targets in IMPORTED library targets in installed
+    <Package>Config.cmake files
+
+    With switch to modern CMake IMPORTED targets, now include directories and
+    critical compiler options and include directories (and other modern CMake
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: "Include directories" is repeated.  That is an obvious error.

File: RELEASE_NOTES [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+    usage requirements) are propagated through linking to the IMPORTED library
+    targets in downstream customer CMake projects.  This makes the usage of
+    the variables `<Trilinos>_INCLUDE_DIRS` and `Trilinos__TPL_INCLUDE_DIRS`
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: There is an extra _ in `Trilinos__TPL_INCLUDE_DIRS`.  That is an obvious typo.

File: RELEASE_NOTES [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+    unnecessary in downstream CMake projects.  However, this change will also
+    cause downstream CMake projects to pull in include directories as `SYSTEM`
```
Comment (bartlettroscoe) [inline RELEASE_NOTES:18]: Should change to:

> as `SYSTEM` *includes*.

File: RELEASE_NOTES [hunk 4]
```diff
@@ -1,3 +1,78 @@
...
+    (e.g. using `-isystem` instead of `-I`) from IMPORTED library targets.
+    This changes how these include directories are searched and could break
+    some fragile build environments that have the same header file names in
+    multiple include directories searched by the compiler.  Also, this will
+    silence any regular compiler warnings from headers found under these
+    include directories.  **This constitutes a break in backward compatibility
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: Where is the matching `**` for the bolding?  That is an obvious error.

File: RELEASE_NOTES [hunk 5]
```diff
@@ -1,3 +1,78 @@
...
+    that will break some customer CMake project builds that uses Trilinos on
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: Obvious typo here.  It should be:

> that will break some customer CMake project builds that *use* Trilinos on

File: RELEASE_NOTES [hunk 6]
```diff
@@ -1,3 +1,78 @@
...
+    fragile environments where the search order of the include directories is
+    important.
+
+    There are several approaches for address this change from `-I` to
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: Here is another obvious typo.  This needs to change to:

> approaches for *addressing* this change

or

> approaches *to* address this change

File: RELEASE_NOTES [hunk 7]
```diff
@@ -1,3 +1,78 @@
...
+    variable:
+
+      -D Trilinos_IMPORTED_NO_SYSTEM=ON
+
+    This will change back the listing of Trilinos include directories in
+    downstream customer CMake projects from `-isystem` and `-I` and will
+    therefore restore about perfect backward compatibility.
+
+    **Approach-2:** Set the cache variable
+    `CMAKE_NO_SYSTEM_FROM_IMPORTED=TRUE` downstream CMake project to set which
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: Obvious typo.  This should be:

> *in a* downstream CMake project

**Review by mperrinel — APPROVED** [review]
I think the problem and the different approaches are well explained.
I didn't see any typo mistake.
Ok for me.

File: RELEASE_NOTES [hunk 8]
```diff
@@ -1,3 +1,78 @@
...
+    these other IMPORTED library targets and may expose many new warnings
+    (which may have been silenced by their include directories being pulled in
+    using `-isystem`).
+
+    **Approach-3:** Clean up the list of include directories that is searched
+    by the compiler so that only the correct header files can be found
+    (regardless of the search order).
+
+    **Approach-4:** Delete some header files in set of searched include
+    directories so that only the correct header files can be found (regardless
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: Another obvious typo.  Could be fixed with:

> Delete some header files in *the* set of searched include directories

File: RELEASE_NOTES [hunk 9]
```diff
@@ -1,3 +1,78 @@
...
+
+    **Approach-5:** Use other approaches more specific to the given customer
+    project.  For example, if multiple different versions of the googletest
+    (gtest) library header files can be found in the list of include
+    directories, then a customer project can build its own version of
+    googletest and put it's include directories first on the compile line and
+    list them with `-I` to ensure that only that version will be found, no
+    matter how many other versions of gtest are installed on the system.  For
+    example, see how this was done for one customer CMake project, see
+    https://github.com/trilinos/Trilinos/issues/8001#issuecomment-1032827124.
```
Comment (bartlettroscoe) [inline RELEASE_NOTES]: Another typo.  Can be fixed with:

> For *an* example *of* how this was done for one customer CMake project, see

**Review by bartlettroscoe — COMMENTED** [review]
I found at least 9 typos or other problems with this text, just from reading it less than 24 hours later.  I a guarantee that I have missed some other problems because I still remember what I think I wrote.

**Comment by bartlettroscoe** [comment]
@marcinwrobel1986 wrote:

> No typos, good to go!

@mperrinel wrote:

> I didn't see any typo mistake.

Guys, a simple rereading of this this morning and I found at least 9 typos or other errors as listed in my review [above](https://github.com/bartlettroscoe/Trilinos/pull/6#pullrequestreview-900505235).

File: RELEASE_NOTES (hunk 1 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Another typo.  Should be:

> With *the* switch to modern CMake

**Comment by bartlettroscoe** [comment]
I fixed the typos and other problems I found [above](https://github.com/bartlettroscoe/Trilinos/pull/6#pullrequestreview-900505235) in the new commit https://github.com/bartlettroscoe/Trilinos/pull/6/commits/6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e that I just pushed.

I think the next step is to have Trilinos developers review this if they want.

File: RELEASE_NOTES (hunk 1 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

File: RELEASE_NOTES (hunk 3 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES:18]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

File: RELEASE_NOTES (hunk 5 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

File: RELEASE_NOTES (hunk 6 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

File: RELEASE_NOTES (hunk 7 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

File: RELEASE_NOTES (hunk 8 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

File: RELEASE_NOTES (hunk 9 above)
Comment (bartlettroscoe) [reply RELEASE_NOTES]: Fixed in 6ab23b5c4825f3a7f8b8ddf6a3a2313d3abe2e9e

**Pull request merged**